addons/mail/wizard/mail_compose_message.py

```diff
@@ -511,18 +510,31 @@ def _process_state(self, mail_values_dict):
             elif optout_emails and mail_to in optout_emails:
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_optout'
-            elif done_emails and mail_to in done_emails and not mailing_document_based:
+            elif mail_to in done_emails:
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_dup'
+            elif mail_to in sent_emails.keys():
+                message_data = {'subject': mail_values['subject'],
+                                'body': mail_values['body'],
+                                'attachment_ids': mail_values['attachment_ids'], }
+                if message_data in sent_emails[mail_to]:
+                    mail_values['state'] = 'cancel'
+                    mail_values['failure_type'] = 'mail_dup'
+                else:
+                    sent_emails[mail_to].append(message_data)
             # void of falsy values -> error
             elif not mail_to:
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_missing'
             elif not mail_to_normalized or not email_re.findall(mail_to):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_invalid'
-            elif done_emails is not None and not mailing_document_based:
-                done_emails.append(mail_to)
+            else:
+                sent_emails[mail_to] = [{'subject': mail_values['subject'],
```

Comment:
As this basically doubles all strings, can't we have sent_emails[mail_to] = [mail_values] ? Normally this will make a link to the original dict, allowing to access it directly from an email without wasting memory ?